hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/contractlog/SyntheticContractLogServiceImpl.java

```diff
@@ -51,16 +52,19 @@ public void create(SyntheticContractLog log) {
 
         ContractLog contractLog = new ContractLog();
 
-        contractLog.setBloom(emptyBloom);
+        contractLog.setBloom(empty);
         contractLog.setConsensusTimestamp(consensusTimestamp);
         contractLog.setContractId(log.getEntityId());
-        contractLog.setData(log.getData());
+        contractLog.setData(log.getData() != null ? log.getData() : empty);
         contractLog.setIndex(logIndex);
         contractLog.setRootContractId(log.getEntityId());
         contractLog.setPayerAccountId(log.getRecordItem().getPayerAccountId());
         contractLog.setTopic0(log.getTopic0());
         contractLog.setTopic1(log.getTopic1());
         contractLog.setTopic2(log.getTopic2());
+        contractLog.setTopic3(log.getTopic3());
+        contractLog.setTransactionIndex(log.getRecordItem().getTransactionIndex());
+        contractLog.setTransactionHash(Arrays.copyOfRange(DomainUtils.toBytes(log.getRecordItem().getTransactionRecord().getTransactionHash()), 0, 32));
```

Comment:
Need to use the logic from `ContractResultServiceImpl` so we get the eth hash, if available. Recommend moving this code from there to a new `RecordItem::getTransactionHash` so it's shared.

```java
public class RecordItem {
    public byte[] getTransactionHash() {
        return Optional.ofNullable(ethereumTransaction)
                .map(EthereumTransaction::getHash)
                .orElseGet(() -> Arrays.copyOfRange(transaction.getTransactionHash(), 0, 32));
    }
```

```suggestion
        contractLog.setTransactionHash(recordItem.getTransactionHash());
```